// @azure/cognitiveservices-contentmoderator/esm/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export var KeyValuePair = {
    serializedName: "KeyValuePair",
    type: {
        name: "Composite",
        className: "KeyValuePair",
        modelProperties: {
            key: {
                serializedName: "Key",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "Value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Tag = {
    serializedName: "Tag",
    type: {
        name: "Composite",
        className: "Tag",
        modelProperties: {
            key: {
                serializedName: "Key",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "Value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Frame = {
    serializedName: "Frame",
    type: {
        name: "Composite",
        className: "Frame",
        modelProperties: {
            timestamp: {
                serializedName: "Timestamp",
                type: {
                    name: "String"
                }
            },
            frameImage: {
                serializedName: "FrameImage",
                type: {
                    name: "String"
                }
            },
            metadata: {
                serializedName: "Metadata",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "KeyValuePair"
                        }
                    }
                }
            },
            reviewerResultTags: {
                serializedName: "ReviewerResultTags",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Tag"
                        }
                    }
                }
            }
        }
    }
};
export var Frames = {
    serializedName: "Frames",
    type: {
        name: "Composite",
        className: "Frames",
        modelProperties: {
            reviewId: {
                serializedName: "ReviewId",
                type: {
                    name: "String"
                }
            },
            videoFrames: {
                serializedName: "VideoFrames",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Frame"
                        }
                    }
                }
            }
        }
    }
};
export var ClassificationCategory1 = {
    serializedName: "Classification_Category1",
    type: {
        name: "Composite",
        className: "ClassificationCategory1",
        modelProperties: {
            score: {
                serializedName: "Score",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var ClassificationCategory2 = {
    serializedName: "Classification_Category2",
    type: {
        name: "Composite",
        className: "ClassificationCategory2",
        modelProperties: {
            score: {
                serializedName: "Score",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var ClassificationCategory3 = {
    serializedName: "Classification_Category3",
    type: {
        name: "Composite",
        className: "ClassificationCategory3",
        modelProperties: {
            score: {
                serializedName: "Score",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var Classification = {
    serializedName: "Classification",
    type: {
        name: "Composite",
        className: "Classification",
        modelProperties: {
            category1: {
                serializedName: "Category1",
                type: {
                    name: "Composite",
                    className: "ClassificationCategory1"
                }
            },
            category2: {
                serializedName: "Category2",
                type: {
                    name: "Composite",
                    className: "ClassificationCategory2"
                }
            },
            category3: {
                serializedName: "Category3",
                type: {
                    name: "Composite",
                    className: "ClassificationCategory3"
                }
            },
            reviewRecommended: {
                serializedName: "ReviewRecommended",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
export var Status = {
    serializedName: "Status",
    type: {
        name: "Composite",
        className: "Status",
        modelProperties: {
            code: {
                serializedName: "Code",
                type: {
                    name: "Number"
                }
            },
            description: {
                serializedName: "Description",
                type: {
                    name: "String"
                }
            },
            exception: {
                serializedName: "Exception",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Email = {
    serializedName: "Email",
    type: {
        name: "Composite",
        className: "Email",
        modelProperties: {
            detected: {
                serializedName: "Detected",
                type: {
                    name: "String"
                }
            },
            subType: {
                serializedName: "SubType",
                type: {
                    name: "String"
                }
            },
            text: {
                serializedName: "Text",
                type: {
                    name: "String"
                }
            },
            index: {
                serializedName: "Index",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var SSN = {
    serializedName: "SSN",
    type: {
        name: "Composite",
        className: "SSN",
        modelProperties: {
            text: {
                serializedName: "Text",
                type: {
                    name: "String"
                }
            },
            index: {
                serializedName: "Index",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var IPA = {
    serializedName: "IPA",
    type: {
        name: "Composite",
        className: "IPA",
        modelProperties: {
            subType: {
                serializedName: "SubType",
                type: {
                    name: "String"
                }
            },
            text: {
                serializedName: "Text",
                type: {
                    name: "String"
                }
            },
            index: {
                serializedName: "Index",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var Phone = {
    serializedName: "Phone",
    type: {
        name: "Composite",
        className: "Phone",
        modelProperties: {
            countryCode: {
                serializedName: "CountryCode",
                type: {
                    name: "String"
                }
            },
            text: {
                serializedName: "Text",
                type: {
                    name: "String"
                }
            },
            index: {
                serializedName: "Index",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var Address = {
    serializedName: "Address",
    type: {
        name: "Composite",
        className: "Address",
        modelProperties: {
            text: {
                serializedName: "Text",
                type: {
                    name: "String"
                }
            },
            index: {
                serializedName: "Index",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var PII = {
    serializedName: "PII",
    type: {
        name: "Composite",
        className: "PII",
        modelProperties: {
            email: {
                serializedName: "Email",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Email"
                        }
                    }
                }
            },
            sSN: {
                serializedName: "SSN",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SSN"
                        }
                    }
                }
            },
            iPA: {
                serializedName: "IPA",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "IPA"
                        }
                    }
                }
            },
            phone: {
                serializedName: "Phone",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Phone"
                        }
                    }
                }
            },
            address: {
                serializedName: "Address",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Address"
                        }
                    }
                }
            }
        }
    }
};
export var DetectedTerms = {
    serializedName: "DetectedTerms",
    type: {
        name: "Composite",
        className: "DetectedTerms",
        modelProperties: {
            index: {
                serializedName: "Index",
                type: {
                    name: "Number"
                }
            },
            originalIndex: {
                serializedName: "OriginalIndex",
                type: {
                    name: "Number"
                }
            },
            listId: {
                serializedName: "ListId",
                type: {
                    name: "Number"
                }
            },
            term: {
                serializedName: "Term",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Screen = {
    serializedName: "Screen",
    type: {
        name: "Composite",
        className: "Screen",
        modelProperties: {
            originalText: {
                serializedName: "OriginalText",
                type: {
                    name: "String"
                }
            },
            normalizedText: {
                serializedName: "NormalizedText",
                type: {
                    name: "String"
                }
            },
            autoCorrectedText: {
                serializedName: "AutoCorrectedText",
                type: {
                    name: "String"
                }
            },
            misrepresentation: {
                serializedName: "Misrepresentation",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            classification: {
                serializedName: "Classification",
                type: {
                    name: "Composite",
                    className: "Classification"
                }
            },
            status: {
                serializedName: "Status",
                type: {
                    name: "Composite",
                    className: "Status"
                }
            },
            pII: {
                serializedName: "PII",
                type: {
                    name: "Composite",
                    className: "PII"
                }
            },
            language: {
                serializedName: "Language",
                type: {
                    name: "String"
                }
            },
            terms: {
                serializedName: "Terms",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DetectedTerms"
                        }
                    }
                }
            },
            trackingId: {
                serializedName: "TrackingId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Face = {
    serializedName: "Face",
    type: {
        name: "Composite",
        className: "Face",
        modelProperties: {
            bottom: {
                serializedName: "Bottom",
                type: {
                    name: "Number"
                }
            },
            left: {
                serializedName: "Left",
                type: {
                    name: "Number"
                }
            },
            right: {
                serializedName: "Right",
                type: {
                    name: "Number"
                }
            },
            top: {
                serializedName: "Top",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var FoundFaces = {
    serializedName: "FoundFaces",
    type: {
        name: "Composite",
        className: "FoundFaces",
        modelProperties: {
            status: {
                serializedName: "Status",
                type: {
                    name: "Composite",
                    className: "Status"
                }
            },
            trackingId: {
                serializedName: "TrackingId",
                type: {
                    name: "String"
                }
            },
            cacheId: {
                serializedName: "CacheId",
                type: {
                    name: "String"
                }
            },
            result: {
                serializedName: "Result",
                type: {
                    name: "Boolean"
                }
            },
            count: {
                serializedName: "Count",
                type: {
                    name: "Number"
                }
            },
            advancedInfo: {
                serializedName: "AdvancedInfo",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "KeyValuePair"
                        }
                    }
                }
            },
            faces: {
                serializedName: "Faces",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Face"
                        }
                    }
                }
            }
        }
    }
};
export var Candidate = {
    serializedName: "Candidate",
    type: {
        name: "Composite",
        className: "Candidate",
        modelProperties: {
            text: {
                serializedName: "Text",
                type: {
                    name: "String"
                }
            },
            confidence: {
                serializedName: "Confidence",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var OCR = {
    serializedName: "OCR",
    type: {
        name: "Composite",
        className: "OCR",
        modelProperties: {
            status: {
                serializedName: "Status",
                type: {
                    name: "Composite",
                    className: "Status"
                }
            },
            metadata: {
                serializedName: "Metadata",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "KeyValuePair"
                        }
                    }
                }
            },
            trackingId: {
                serializedName: "TrackingId",
                type: {
                    name: "String"
                }
            },
            cacheId: {
                serializedName: "CacheId",
                type: {
                    name: "String"
                }
            },
            language: {
                serializedName: "Language",
                type: {
                    name: "String"
                }
            },
            text: {
                serializedName: "Text",
                type: {
                    name: "String"
                }
            },
            candidates: {
                serializedName: "Candidates",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Candidate"
                        }
                    }
                }
            }
        }
    }
};
export var Evaluate = {
    serializedName: "Evaluate",
    type: {
        name: "Composite",
        className: "Evaluate",
        modelProperties: {
            cacheID: {
                serializedName: "CacheID",
                type: {
                    name: "String"
                }
            },
            result: {
                serializedName: "Result",
                type: {
                    name: "Boolean"
                }
            },
            trackingId: {
                serializedName: "TrackingId",
                type: {
                    name: "String"
                }
            },
            adultClassificationScore: {
                serializedName: "AdultClassificationScore",
                type: {
                    name: "Number"
                }
            },
            isImageAdultClassified: {
                serializedName: "IsImageAdultClassified",
                type: {
                    name: "Boolean"
                }
            },
            racyClassificationScore: {
                serializedName: "RacyClassificationScore",
                type: {
                    name: "Number"
                }
            },
            isImageRacyClassified: {
                serializedName: "IsImageRacyClassified",
                type: {
                    name: "Boolean"
                }
            },
            advancedInfo: {
                serializedName: "AdvancedInfo",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "KeyValuePair"
                        }
                    }
                }
            },
            status: {
                serializedName: "Status",
                type: {
                    name: "Composite",
                    className: "Status"
                }
            }
        }
    }
};
export var Match = {
    serializedName: "Match",
    type: {
        name: "Composite",
        className: "Match",
        modelProperties: {
            score: {
                serializedName: "Score",
                type: {
                    name: "Number"
                }
            },
            matchId: {
                serializedName: "MatchId",
                type: {
                    name: "Number"
                }
            },
            source: {
                serializedName: "Source",
                type: {
                    name: "String"
                }
            },
            tags: {
                serializedName: "Tags",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Number"
                        }
                    }
                }
            },
            label: {
                serializedName: "Label",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var MatchResponse = {
    serializedName: "MatchResponse",
    type: {
        name: "Composite",
        className: "MatchResponse",
        modelProperties: {
            trackingId: {
                serializedName: "TrackingId",
                type: {
                    name: "String"
                }
            },
            cacheID: {
                serializedName: "CacheID",
                type: {
                    name: "String"
                }
            },
            isMatch: {
                serializedName: "IsMatch",
                type: {
                    name: "Boolean"
                }
            },
            matches: {
                serializedName: "Matches",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Match"
                        }
                    }
                }
            },
            status: {
                serializedName: "Status",
                type: {
                    name: "Composite",
                    className: "Status"
                }
            }
        }
    }
};
export var DetectedLanguage = {
    serializedName: "DetectedLanguage",
    type: {
        name: "Composite",
        className: "DetectedLanguage",
        modelProperties: {
            detectedLanguage: {
                serializedName: "DetectedLanguage",
                type: {
                    name: "String"
                }
            },
            status: {
                serializedName: "Status",
                type: {
                    name: "Composite",
                    className: "Status"
                }
            },
            trackingId: {
                serializedName: "TrackingId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ImageList = {
    serializedName: "ImageList",
    type: {
        name: "Composite",
        className: "ImageList",
        modelProperties: {
            id: {
                serializedName: "Id",
                type: {
                    name: "Number"
                }
            },
            name: {
                serializedName: "Name",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "Description",
                type: {
                    name: "String"
                }
            },
            metadata: {
                serializedName: "Metadata",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export var TermList = {
    serializedName: "TermList",
    type: {
        name: "Composite",
        className: "TermList",
        modelProperties: {
            id: {
                serializedName: "Id",
                type: {
                    name: "Number"
                }
            },
            name: {
                serializedName: "Name",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "Description",
                type: {
                    name: "String"
                }
            },
            metadata: {
                serializedName: "Metadata",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export var RefreshIndex = {
    serializedName: "RefreshIndex",
    type: {
        name: "Composite",
        className: "RefreshIndex",
        modelProperties: {
            contentSourceId: {
                serializedName: "ContentSourceId",
                type: {
                    name: "String"
                }
            },
            isUpdateSuccess: {
                serializedName: "IsUpdateSuccess",
                type: {
                    name: "Boolean"
                }
            },
            advancedInfo: {
                serializedName: "AdvancedInfo",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Dictionary",
                            value: {
                                type: {
                                    name: "String"
                                }
                            }
                        }
                    }
                }
            },
            status: {
                serializedName: "Status",
                type: {
                    name: "Composite",
                    className: "Status"
                }
            },
            trackingId: {
                serializedName: "TrackingId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ImageAdditionalInfoItem = {
    serializedName: "Image_AdditionalInfoItem",
    type: {
        name: "Composite",
        className: "ImageAdditionalInfoItem",
        modelProperties: {
            key: {
                serializedName: "Key",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "Value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Image = {
    serializedName: "Image",
    type: {
        name: "Composite",
        className: "Image",
        modelProperties: {
            contentId: {
                serializedName: "ContentId",
                type: {
                    name: "String"
                }
            },
            additionalInfo: {
                serializedName: "AdditionalInfo",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ImageAdditionalInfoItem"
                        }
                    }
                }
            },
            status: {
                serializedName: "Status",
                type: {
                    name: "Composite",
                    className: "Status"
                }
            },
            trackingId: {
                serializedName: "TrackingId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ImageIds = {
    serializedName: "ImageIds",
    type: {
        name: "Composite",
        className: "ImageIds",
        modelProperties: {
            contentSource: {
                serializedName: "ContentSource",
                type: {
                    name: "String"
                }
            },
            contentIds: {
                serializedName: "ContentIds",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Number"
                        }
                    }
                }
            },
            status: {
                serializedName: "Status",
                type: {
                    name: "Composite",
                    className: "Status"
                }
            },
            trackingId: {
                serializedName: "TrackingId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TermsInList = {
    serializedName: "TermsInList",
    type: {
        name: "Composite",
        className: "TermsInList",
        modelProperties: {
            term: {
                serializedName: "Term",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TermsData = {
    serializedName: "TermsData",
    type: {
        name: "Composite",
        className: "TermsData",
        modelProperties: {
            language: {
                serializedName: "Language",
                type: {
                    name: "String"
                }
            },
            terms: {
                serializedName: "Terms",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "TermsInList"
                        }
                    }
                }
            },
            status: {
                serializedName: "Status",
                type: {
                    name: "Composite",
                    className: "Status"
                }
            },
            trackingId: {
                serializedName: "TrackingId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TermsPaging = {
    serializedName: "Terms_Paging",
    type: {
        name: "Composite",
        className: "TermsPaging",
        modelProperties: {
            total: {
                serializedName: "Total",
                type: {
                    name: "Number"
                }
            },
            limit: {
                serializedName: "Limit",
                type: {
                    name: "Number"
                }
            },
            offset: {
                serializedName: "Offset",
                type: {
                    name: "Number"
                }
            },
            returned: {
                serializedName: "Returned",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var Terms = {
    serializedName: "Terms",
    type: {
        name: "Composite",
        className: "Terms",
        modelProperties: {
            data: {
                serializedName: "Data",
                type: {
                    name: "Composite",
                    className: "TermsData"
                }
            },
            paging: {
                serializedName: "Paging",
                type: {
                    name: "Composite",
                    className: "TermsPaging"
                }
            }
        }
    }
};
export var Review = {
    serializedName: "Review",
    type: {
        name: "Composite",
        className: "Review",
        modelProperties: {
            reviewId: {
                serializedName: "ReviewId",
                type: {
                    name: "String"
                }
            },
            subTeam: {
                serializedName: "SubTeam",
                type: {
                    name: "String"
                }
            },
            status: {
                serializedName: "Status",
                type: {
                    name: "String"
                }
            },
            reviewerResultTags: {
                serializedName: "ReviewerResultTags",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "KeyValuePair"
                        }
                    }
                }
            },
            createdBy: {
                serializedName: "CreatedBy",
                type: {
                    name: "String"
                }
            },
            metadata: {
                serializedName: "Metadata",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "KeyValuePair"
                        }
                    }
                }
            },
            type: {
                serializedName: "Type",
                type: {
                    name: "String"
                }
            },
            content: {
                serializedName: "Content",
                type: {
                    name: "String"
                }
            },
            contentId: {
                serializedName: "ContentId",
                type: {
                    name: "String"
                }
            },
            callbackEndpoint: {
                serializedName: "CallbackEndpoint",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var JobExecutionReportDetails = {
    serializedName: "JobExecutionReportDetails",
    type: {
        name: "Composite",
        className: "JobExecutionReportDetails",
        modelProperties: {
            ts: {
                serializedName: "Ts",
                type: {
                    name: "String"
                }
            },
            msg: {
                serializedName: "Msg",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var Job = {
    serializedName: "Job",
    type: {
        name: "Composite",
        className: "Job",
        modelProperties: {
            id: {
                serializedName: "Id",
                type: {
                    name: "String"
                }
            },
            teamName: {
                serializedName: "TeamName",
                type: {
                    name: "String"
                }
            },
            status: {
                serializedName: "Status",
                type: {
                    name: "String"
                }
            },
            workflowId: {
                serializedName: "WorkflowId",
                type: {
                    name: "String"
                }
            },
            type: {
                serializedName: "Type",
                type: {
                    name: "String"
                }
            },
            callBackEndpoint: {
                serializedName: "CallBackEndpoint",
                type: {
                    name: "String"
                }
            },
            reviewId: {
                serializedName: "ReviewId",
                type: {
                    name: "String"
                }
            },
            resultMetaData: {
                serializedName: "ResultMetaData",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "KeyValuePair"
                        }
                    }
                }
            },
            jobExecutionReport: {
                serializedName: "JobExecutionReport",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "JobExecutionReportDetails"
                        }
                    }
                }
            }
        }
    }
};
export var JobListResult = {
    serializedName: "JobListResult",
    type: {
        name: "Composite",
        className: "JobListResult",
        modelProperties: {
            value: {
                serializedName: "Value",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export var JobId = {
    serializedName: "JobId",
    type: {
        name: "Composite",
        className: "JobId",
        modelProperties: {
            jobId: {
                serializedName: "JobId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ErrorModel = {
    serializedName: "Error",
    type: {
        name: "Composite",
        className: "ErrorModel",
        modelProperties: {
            code: {
                serializedName: "Code",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "Message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var APIError = {
    serializedName: "APIError",
    type: {
        name: "Composite",
        className: "APIError",
        modelProperties: {
            error: {
                serializedName: "Error",
                type: {
                    name: "Composite",
                    className: "ErrorModel"
                }
            }
        }
    }
};
export var Body = {
    serializedName: "body",
    type: {
        name: "Composite",
        className: "Body",
        modelProperties: {
            name: {
                serializedName: "Name",
                type: {
                    name: "String"
                }
            },
            description: {
                serializedName: "Description",
                type: {
                    name: "String"
                }
            },
            metadata: {
                serializedName: "Metadata",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export var CreateReviewBodyItemMetadataItem = {
    serializedName: "createReviewBodyItem_MetadataItem",
    type: {
        name: "Composite",
        className: "CreateReviewBodyItemMetadataItem",
        modelProperties: {
            key: {
                required: true,
                serializedName: "Key",
                type: {
                    name: "String"
                }
            },
            value: {
                required: true,
                serializedName: "Value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var CreateReviewBodyItem = {
    serializedName: "createReviewBodyItem",
    type: {
        name: "Composite",
        className: "CreateReviewBodyItem",
        modelProperties: {
            type: {
                required: true,
                serializedName: "Type",
                type: {
                    name: "String"
                }
            },
            content: {
                required: true,
                serializedName: "Content",
                type: {
                    name: "String"
                }
            },
            contentId: {
                required: true,
                serializedName: "ContentId",
                type: {
                    name: "String"
                }
            },
            callbackEndpoint: {
                serializedName: "CallbackEndpoint",
                type: {
                    name: "String"
                }
            },
            metadata: {
                serializedName: "Metadata",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CreateReviewBodyItemMetadataItem"
                        }
                    }
                }
            }
        }
    }
};
export var Content = {
    serializedName: "Content",
    type: {
        name: "Composite",
        className: "Content",
        modelProperties: {
            contentValue: {
                required: true,
                serializedName: "ContentValue",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TranscriptModerationBodyItemTermsItem = {
    serializedName: "transcriptModerationBodyItem_TermsItem",
    type: {
        name: "Composite",
        className: "TranscriptModerationBodyItemTermsItem",
        modelProperties: {
            index: {
                required: true,
                serializedName: "Index",
                type: {
                    name: "Number"
                }
            },
            term: {
                required: true,
                serializedName: "Term",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TranscriptModerationBodyItem = {
    serializedName: "transcriptModerationBodyItem",
    type: {
        name: "Composite",
        className: "TranscriptModerationBodyItem",
        modelProperties: {
            timestamp: {
                required: true,
                serializedName: "Timestamp",
                type: {
                    name: "String"
                }
            },
            terms: {
                required: true,
                serializedName: "Terms",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "TranscriptModerationBodyItemTermsItem"
                        }
                    }
                }
            }
        }
    }
};
export var ImageUrl = {
    serializedName: "ImageUrl",
    type: {
        name: "Composite",
        className: "ImageUrl",
        modelProperties: {
            dataRepresentation: {
                serializedName: "DataRepresentation",
                defaultValue: 'URL',
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "Value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var CreateVideoReviewsBodyItemVideoFramesItemReviewerResultTagsItem = {
    serializedName: "CreateVideoReviewsBodyItem_VideoFramesItem_ReviewerResultTagsItem",
    type: {
        name: "Composite",
        className: "CreateVideoReviewsBodyItemVideoFramesItemReviewerResultTagsItem",
        modelProperties: {
            key: {
                required: true,
                serializedName: "Key",
                type: {
                    name: "String"
                }
            },
            value: {
                required: true,
                serializedName: "Value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var CreateVideoReviewsBodyItemVideoFramesItemMetadataItem = {
    serializedName: "CreateVideoReviewsBodyItem_VideoFramesItem_MetadataItem",
    type: {
        name: "Composite",
        className: "CreateVideoReviewsBodyItemVideoFramesItemMetadataItem",
        modelProperties: {
            key: {
                required: true,
                serializedName: "Key",
                type: {
                    name: "String"
                }
            },
            value: {
                required: true,
                serializedName: "Value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var CreateVideoReviewsBodyItemVideoFramesItem = {
    serializedName: "CreateVideoReviewsBodyItem_VideoFramesItem",
    type: {
        name: "Composite",
        className: "CreateVideoReviewsBodyItemVideoFramesItem",
        modelProperties: {
            id: {
                required: true,
                serializedName: "Id",
                type: {
                    name: "String"
                }
            },
            timestamp: {
                required: true,
                serializedName: "Timestamp",
                type: {
                    name: "Number"
                }
            },
            frameImage: {
                required: true,
                serializedName: "FrameImage",
                type: {
                    name: "String"
                }
            },
            reviewerResultTags: {
                serializedName: "ReviewerResultTags",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CreateVideoReviewsBodyItemVideoFramesItemReviewerResultTagsItem"
                        }
                    }
                }
            },
            metadata: {
                serializedName: "Metadata",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CreateVideoReviewsBodyItemVideoFramesItemMetadataItem"
                        }
                    }
                }
            }
        }
    }
};
export var CreateVideoReviewsBodyItemMetadataItem = {
    serializedName: "CreateVideoReviewsBodyItem_MetadataItem",
    type: {
        name: "Composite",
        className: "CreateVideoReviewsBodyItemMetadataItem",
        modelProperties: {
            key: {
                required: true,
                serializedName: "Key",
                type: {
                    name: "String"
                }
            },
            value: {
                required: true,
                serializedName: "Value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var CreateVideoReviewsBodyItem = {
    serializedName: "CreateVideoReviewsBodyItem",
    type: {
        name: "Composite",
        className: "CreateVideoReviewsBodyItem",
        modelProperties: {
            videoFrames: {
                serializedName: "VideoFrames",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CreateVideoReviewsBodyItemVideoFramesItem"
                        }
                    }
                }
            },
            metadata: {
                serializedName: "Metadata",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CreateVideoReviewsBodyItemMetadataItem"
                        }
                    }
                }
            },
            type: {
                required: true,
                isConstant: true,
                serializedName: "Type",
                defaultValue: 'Video',
                type: {
                    name: "String"
                }
            },
            content: {
                required: true,
                serializedName: "Content",
                type: {
                    name: "String"
                }
            },
            contentId: {
                required: true,
                serializedName: "ContentId",
                type: {
                    name: "String"
                }
            },
            status: {
                required: true,
                serializedName: "Status",
                type: {
                    name: "String"
                }
            },
            timescale: {
                serializedName: "Timescale",
                type: {
                    name: "Number"
                }
            },
            callbackEndpoint: {
                serializedName: "CallbackEndpoint",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var VideoFrameBodyItemReviewerResultTagsItem = {
    serializedName: "videoFrameBodyItem_ReviewerResultTagsItem",
    type: {
        name: "Composite",
        className: "VideoFrameBodyItemReviewerResultTagsItem",
        modelProperties: {
            key: {
                required: true,
                serializedName: "Key",
                type: {
                    name: "String"
                }
            },
            value: {
                required: true,
                serializedName: "Value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var VideoFrameBodyItemMetadataItem = {
    serializedName: "videoFrameBodyItem_MetadataItem",
    type: {
        name: "Composite",
        className: "VideoFrameBodyItemMetadataItem",
        modelProperties: {
            key: {
                required: true,
                serializedName: "Key",
                type: {
                    name: "String"
                }
            },
            value: {
                required: true,
                serializedName: "Value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var VideoFrameBodyItem = {
    serializedName: "videoFrameBodyItem",
    type: {
        name: "Composite",
        className: "VideoFrameBodyItem",
        modelProperties: {
            timestamp: {
                required: true,
                serializedName: "Timestamp",
                type: {
                    name: "String"
                }
            },
            frameImage: {
                required: true,
                serializedName: "FrameImage",
                type: {
                    name: "String"
                }
            },
            reviewerResultTags: {
                serializedName: "ReviewerResultTags",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VideoFrameBodyItemReviewerResultTagsItem"
                        }
                    }
                }
            },
            metadata: {
                serializedName: "Metadata",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VideoFrameBodyItemMetadataItem"
                        }
                    }
                }
            }
        }
    }
};
