// @azure/cognitiveservices-contentmoderator/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var autocorrect = {
    parameterPath: [
        "options",
        "autocorrect"
    ],
    mapper: {
        serializedName: "autocorrect",
        defaultValue: false,
        type: {
            name: "Boolean"
        }
    }
};
export var cacheImage = {
    parameterPath: [
        "options",
        "cacheImage"
    ],
    mapper: {
        serializedName: "CacheImage",
        type: {
            name: "Boolean"
        }
    }
};
export var callBackEndpoint = {
    parameterPath: [
        "options",
        "callBackEndpoint"
    ],
    mapper: {
        serializedName: "CallBackEndpoint",
        type: {
            name: "String"
        }
    }
};
export var classify = {
    parameterPath: [
        "options",
        "classify"
    ],
    mapper: {
        serializedName: "classify",
        defaultValue: false,
        type: {
            name: "Boolean"
        }
    }
};
export var contentId = {
    parameterPath: "contentId",
    mapper: {
        required: true,
        serializedName: "ContentId",
        type: {
            name: "String"
        }
    }
};
export var contentType0 = {
    parameterPath: "contentType",
    mapper: {
        required: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export var contentType1 = {
    parameterPath: "contentType",
    mapper: {
        required: true,
        serializedName: "ContentType",
        type: {
            name: "String"
        }
    }
};
export var contentType2 = {
    parameterPath: "contentType",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "Content-Type",
        defaultValue: 'text/plain',
        type: {
            name: "String"
        }
    }
};
export var endpoint = {
    parameterPath: "endpoint",
    mapper: {
        required: true,
        serializedName: "Endpoint",
        defaultValue: '',
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var enhanced = {
    parameterPath: [
        "options",
        "enhanced"
    ],
    mapper: {
        serializedName: "enhanced",
        defaultValue: false,
        type: {
            name: "Boolean"
        }
    }
};
export var filter = {
    parameterPath: [
        "options",
        "filter"
    ],
    mapper: {
        serializedName: "filter",
        type: {
            name: "String"
        }
    }
};
export var frameImageZip = {
    parameterPath: "frameImageZip",
    mapper: {
        required: true,
        serializedName: "frameImageZip",
        type: {
            name: "Stream"
        }
    }
};
export var frameMetadata = {
    parameterPath: "frameMetadata",
    mapper: {
        required: true,
        serializedName: "frameMetadata",
        type: {
            name: "String"
        }
    }
};
export var imageId = {
    parameterPath: "imageId",
    mapper: {
        required: true,
        serializedName: "ImageId",
        type: {
            name: "String"
        }
    }
};
export var jobContentType = {
    parameterPath: "jobContentType",
    mapper: {
        required: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export var jobId = {
    parameterPath: "jobId",
    mapper: {
        required: true,
        serializedName: "JobId",
        type: {
            name: "String"
        }
    }
};
export var label = {
    parameterPath: [
        "options",
        "label"
    ],
    mapper: {
        serializedName: "label",
        type: {
            name: "String"
        }
    }
};
export var language0 = {
    parameterPath: "language",
    mapper: {
        required: true,
        serializedName: "language",
        type: {
            name: "String"
        }
    }
};
export var language1 = {
    parameterPath: [
        "options",
        "language"
    ],
    mapper: {
        serializedName: "language",
        type: {
            name: "String"
        }
    }
};
export var limit = {
    parameterPath: [
        "options",
        "limit"
    ],
    mapper: {
        serializedName: "limit",
        type: {
            name: "Number"
        }
    }
};
export var listId0 = {
    parameterPath: [
        "options",
        "listId"
    ],
    mapper: {
        serializedName: "listId",
        type: {
            name: "String"
        }
    }
};
export var listId1 = {
    parameterPath: "listId",
    mapper: {
        required: true,
        serializedName: "listId",
        type: {
            name: "String"
        }
    }
};
export var noOfRecords = {
    parameterPath: [
        "options",
        "noOfRecords"
    ],
    mapper: {
        serializedName: "noOfRecords",
        type: {
            name: "Number"
        }
    }
};
export var offset = {
    parameterPath: [
        "options",
        "offset"
    ],
    mapper: {
        serializedName: "offset",
        type: {
            name: "Number"
        }
    }
};
export var pII = {
    parameterPath: [
        "options",
        "pII"
    ],
    mapper: {
        serializedName: "PII",
        defaultValue: false,
        type: {
            name: "Boolean"
        }
    }
};
export var reviewId = {
    parameterPath: "reviewId",
    mapper: {
        required: true,
        serializedName: "reviewId",
        type: {
            name: "String"
        }
    }
};
export var startSeed = {
    parameterPath: [
        "options",
        "startSeed"
    ],
    mapper: {
        serializedName: "startSeed",
        type: {
            name: "Number"
        }
    }
};
export var subTeam = {
    parameterPath: [
        "options",
        "subTeam"
    ],
    mapper: {
        serializedName: "subTeam",
        type: {
            name: "String"
        }
    }
};
export var tag = {
    parameterPath: [
        "options",
        "tag"
    ],
    mapper: {
        serializedName: "tag",
        type: {
            name: "Number"
        }
    }
};
export var teamName = {
    parameterPath: "teamName",
    mapper: {
        required: true,
        serializedName: "teamName",
        type: {
            name: "String"
        }
    }
};
export var term = {
    parameterPath: "term",
    mapper: {
        required: true,
        serializedName: "term",
        type: {
            name: "String"
        }
    }
};
export var textContentType = {
    parameterPath: "textContentType",
    mapper: {
        required: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export var timescale = {
    parameterPath: [
        "options",
        "timescale"
    ],
    mapper: {
        serializedName: "timescale",
        type: {
            name: "Number"
        }
    }
};
export var urlContentType = {
    parameterPath: "urlContentType",
    mapper: {
        required: true,
        serializedName: "UrlContentType",
        type: {
            name: "String"
        }
    }
};
export var workflowName = {
    parameterPath: "workflowName",
    mapper: {
        required: true,
        serializedName: "WorkflowName",
        type: {
            name: "String"
        }
    }
};
